applications/minotari_node/src/bootstrap.rs

```diff
@@ -164,10 +172,27 @@ where B: BlockchainBackend + 'static
 
         let comms = comms.add_protocol_extension(mempool_protocol);
         let comms = Self::setup_rpc_services(comms, &handles, self.db.into(), &p2p_config);
-        let comms = initialization::spawn_comms_using_transport(comms, p2p_config.transport.clone())
-            .await
-            .map_err(|e| e.to_exit_error())?;
 
+        let comms = if p2p_config.transport.transport_type == TransportType::Tor {
+            let path = base_node_config.tor_identity_file.clone();
+            let node_id = comms.node_identity();
+            let after_comms = move |identity: TorIdentity| {
+                let _result = identity_management::save_as_json(&path, &identity);
+                trace!(target: LOG_TARGET, "resave the tor identity {:?}", identity);
+                let address: Multiaddr = format!("/onion3/{}:{}", identity.service_id, identity.onion_port)
+                    .parse()
+                    .expect("Should be able to create address");
```

Comment:
This expect could cause a bunch of problems. Perhaps better to log a warning and continue?